spec/controllers/api/v1/books_controller_spec.rb

```diff
@@ -1,61 +1,54 @@
 require 'rails_helper'
 
-describe Api::V1::BooksController, type: :controller do
+describe Api::V1::BooksController do
   include_context 'with Authenticated User'
 
   describe 'GET #index' do
+    subject(:http_request) { get :index }
+
     context 'when fetching all the books' do
       let(:books) { create_list(:book, 3) }
 
-      before do
-        get :index
-      end
-
       it 'reponses with books json' do
         expected = ActiveModel::Serializer::CollectionSerializer.new(
           books, each_serializer: BookSerializer
         ).to_json
-        expect(response.body.to_json) =~ JSON.parse(expected)
+        expect(http_request.body.to_json) =~ JSON.parse(expected)
```

Comment:
the call to the `http_request` method can be made in a `before...end` block and use the `response_body` method to check the response body directly

Reply:
Done